zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/InterpreterContext.java

```diff
@@ -27,6 +27,21 @@
  * Interpreter context
  */
 public class InterpreterContext {
+  private static final ThreadLocal<InterpreterContext> threadIC =
```

Comment:
Would it be too big change to make `InterpreterContext` a singleton, and then add non-static field of the `TreadLocal`?